### PR TITLE
Added Simple Test 1

### PR DESCRIPTION
**Pull request by RalucaMihaela**
I added the solving of the Simple Test 1 and 2.

### CODE REVIEW

**Review by drqCode — COMMENTED** [review]
Why are all the files included in the repository like node modules? You should use .gitignore: https://www.atlassian.com/git/tutorials/saving-changes/gitignore.

**Comment by RalucaMihaela** [comment]
I included all the files, because on some of my projects I had the situation where my app broke because of bad dependencies. Maybe in time the packages will change and if the app depends on npm module it can't be updated anymore.
	I think that this depends on the type of the project. I will use .gitignore to remain only the necessary files.

**Comment by drqCode** [comment]
node_modules don't need to be included. In case of installing wrong dependencies with npm, there is package.lock file (it has to be added to git). See https://docs.npmjs.com/files/package-lock.json. :+1:

**Comment by alexandru-calinoiu** [comment]
@RalucaMihaela the proper way to handle node modules is to use a `package.json` and the latest version of `npm` will generate a `package-lock.json` that will make sure the exact version of packages is installed on all machines.

We can ignore this issue for now.

**Comment by alexandru-calinoiu** [comment]
For your consideration here are a couple of general guidelines:

- use js conventions, for example file names by conventions use - not camel case, https://stackoverflow.com/questions/7273316/what-is-the-javascript-filename-naming-convention
- don't 'overthink' simple solutions, pb1 states `given an array of integers` there is no reason for you to guard against floats or any other type of character.

**Comment by RalucaMihaela** [comment]
I understand. I will use js naming convensions.  I tested for floats or other type of characters, because I thought that if I have an array of integers I have to make sure there are only integers elements.

File: duplicates-in-array.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = function (array, n){
+  var hasDuplicates = false;
+  var sum = 0;
+  var length = array.length;
+    if ( array.length === 0 ) return false;
+  	if( n != length ) return false;
```
Comment (alexandru-calinoiu) [inline duplicates-in-array.js:6]: The code indentation here makes it really hard to read.

File: duplicates-test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      it('If there are duplicates return true', function(){
+          var array = [1,4,5,2,4];
+          var n = 5;
+          assert.equal(app(array,n),true);
+      });
+      it('If there are not duplicates return false', function(){
+          var array = [1,4,5,2,3];
+          var n = 5;
+          assert.equal(app(array,n),false);
+      });
```
Comment (alexandru-calinoiu) [inline duplicates-test.js:41]: How about a test for `[2, 2, 2]` ?

File: simpleTest2/appSecondPb.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+module.exports = function (array) {
+
+  	var firstMAX = Number.NEGATIVE_INFINITY;
+    var secondMAX = Number.NEGATIVE_INFINITY;
+
+    if(array.length === 0) return 0;
+
+    for(var index = 0; index < array.length; index++) {
+            if(array.length === 1) return [array[index],0];
```
Comment (alexandru-calinoiu) [inline simpleTest2/appSecondPb.js:9]: This can be pulled up.

File: simpleTest1/appFirstPb.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+module.exports = function (array) {
+
+    var sum = 0;
+    let notNumber = 0;
+
+    if(array.length > 0) {
```
Comment (alexandru-calinoiu) [inline simpleTest1/appFirstPb.js:6]: Do you really need this here, what will happen if you don't have it and the array passed is `[]`?

**Review by alexandru-calinoiu — CHANGES_REQUESTED** [review]
Apart from the inline comments related to problem 3:

- the form of the code is bad, code indentation is missing and it looks sloppy
- added to many files to the repo
 
On the good side:

- the tests are good and testing as many cases as possible.

File: duplicates-in-array.js (hunk 1 above)
Comment (RalucaMihaela) [reply duplicates-in-array.js:6]: Yes, I see. Usually I am more careful with this. In my files  those two conditions are aligned.

File: duplicates-test.js (hunk 2 above)
Comment (RalucaMihaela) [reply duplicates-test.js:41]: Yes, for this test my code returns that there are no duplicates. I tested for situations where I have numbers from 1 to n.  In this case I can verify if there are duplicates using another array. In this array I will store the original array and I will compare each item with the original array. I do not know how to do this only with one loop and without another structure.

File: duplicates-in-array.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+module.exports = function (array, n){
```
Comment (drqCode) [inline duplicates-in-array.js:1]: Why do you need `n` as parameter? The problem description says: "Given an array of numbers having size N in which every number is between 1 and N". In my opinion, it doesn't state that you'll be given an array of numbers and a number representing the size or the number of elements.

File: simpleTest2/appSecondPb.js (hunk 3 above)
Comment (RalucaMihaela) [reply simpleTest2/appSecondPb.js:9]: Now I see that I do not need this as I am already printing [1,-Infinity] if I have just one item like [1].

File: simpleTest1/appFirstPb.js (hunk 4 above)
Comment (RalucaMihaela) [reply simpleTest1/appFirstPb.js:6]: If I don't have it, the sum will be 0, so yes I do not need that condition there.

File: duplicates-in-array.js (hunk 5 above)
Comment (RalucaMihaela) [reply duplicates-in-array.js:1]: Yes it does not say, but I used n to verify if my given array has the number of items I want and to verify if each item is smaller than n.  I could have used only the array's length to verify if each number is smaller than the length.

Comment (drqCode) [reply duplicates-in-array.js:1]: Given the description, you do not need to check if the array is a good one (that's others responsibility), you only have to check for duplicates. :smiley:

**Comment by alexandru-calinoiu** [comment]
Closing this as it appears it has been replace by https://github.com/Agilefreaks/Test/pull/26

File: duplicates-test.js (hunk 2 above)
Comment (alexandru-calinoiu) [reply duplicates-test.js:41]: Not sure what you mean, [2,2,2] is between 1..n

There is a way to solve this problem with only one datastructure, notice that you can update/destroy the existing array.

**Pull request closed without merging**